### PR TITLE
Fixes provisioning output for container app deployments

### PR DESCRIPTION
**Pull request by wbreza**
Since container app deployments today use a resource group scoped Bicep deployment the output required some updates to correctly display the effected resources.

- [x] Updates resource manager to leverage `infra.Scope`
- [x] Adds support in `infra.Scope` for both subscription level and resource group deployments
- [x] Moves `Scope` from `provisioning` package to `infra` package.

### CODE REVIEW

File: cli/azd/pkg/project/service_target_containerapp.go [hunk 1]
```diff
@@ -42,7 +43,7 @@ func (at *containerAppTarget) Deploy(ctx context.Context, azdCtx *azdcontext.Azd
 		at.config.Infra.Module = at.config.Name
 	}
 
-	infraManager, err := provisioning.NewManager(ctx, *at.env, at.config.Project.Path, at.config.Infra, false)
+	infraManager, err := provisioning.NewManager(ctx, *at.env, at.config.Project.Path, at.config.Infra, true)
```
Comment (weikanglim) [inline cli/azd/pkg/project/service_target_containerapp.go:46]: This hardcoded value makes me wonder if the `interactive` options isn't passed correctly, or that provisioning manager shouldn't know about interactivity if it's meant to be used in this way..

File: cli/azd/pkg/infra/provisioning/provisioning_progress_display.go [hunk 2]
```diff
@@ -23,31 +23,29 @@ const succeededProvisioningState string = "Succeeded"
...
-		deploymentName:   deploymentName,
+		scope:            scope,
 		resourceManager:  rm,
 		console:          console,
 	}
 }
 
 // ReportProgress reports the current deployment progress, setting the currently executing operation title and logging progress.
-func (display *ProvisioningProgressDisplay) ReportProgress(ctx context.Context) (*DeployProgress, error) {
+func (display *ProvisioningProgressDisplay) ReportProgress(ctx context.Context, scope infra.Scope) (*DeployProgress, error) {
```
Comment (weikanglim) [inline cli/azd/pkg/infra/provisioning/provisioning_progress_display.go:41]: Does `scope` still need to be passed to `ReportProgress` if it's also part stored in the struct? I think the lifetime should be with the object (at least the way it was written initially)

File: cli/azd/pkg/infra/azure_resource_manager.go [hunk 3]
```diff
@@ -54,6 +54,13 @@ func (rm *AzureResourceManager) GetDeploymentResourceOperations(ctx context.Cont
 		}
 	}
 
+	if resourceGroupName == "" {
+		for _, operation := range subOperations {
+			resourceGroupName = operation.Properties.TargetResource.ResourceGroup
+			break
+		}
+	}
```
Comment (weikanglim) [inline cli/azd/pkg/infra/azure_resource_manager.go:62]: Is this addressing a new issue that occurs with resource group scopes? Is it better to have explicit conditional handling based on the type of scope? For example, if scope is resource group scope, I think the resource group name would be provided directly by the scope.

The fallback seems a little unintuitive unless we can codify in comments how it applies to the general case.

File: cli/azd/pkg/infra/azure_resource_manager.go [hunk 4]
```diff
@@ -64,13 +71,17 @@ func (rm *AzureResourceManager) GetDeploymentResourceOperations(ctx context.Cont
...
+			err = rm.appendDeploymentResourcesRecursive(ctx, scope.SubscriptionId(), resourceGroupName, operation.Properties.TargetResource.ResourceName, &resourceOperations)
 			if err != nil {
 				return nil, fmt.Errorf("appending deployment resources: %w", err)
 			}
 		}
 	}
 
+	if len(resourceOperations) == 0 {
+		resourceOperations = subOperations
+	}
```
Comment (weikanglim) [inline cli/azd/pkg/infra/azure_resource_manager.go:83]: I think this is similarly unclear why it exists. I suspect it is related to handling resource group scopes vs. subscription scopes and I wonder if we should make it part of an explicit scope-type handling

**Review by weikanglim — CHANGES_REQUESTED** [review]
Everything looks good... Just want a quick discussion on the `GetDeploymentResourceOperations` handling before final approval.

**Comment by wbreza** [comment]
> Everything looks good... Just want a quick discussion on the `GetDeploymentResourceOperations` handling before final approval.

Thanks for the great feedback @weikanglim.  Everything should now be addressed.

**Review by weikanglim — APPROVED** [review]
Thanks for addressing all the comments. Looks good to me!

**Pull request merged**